src/class/net/net_device.c

```diff
@@ -226,33 +249,89 @@ static void ecm_report(bool nc)
 // return false to stall control endpoint (e.g unsupported request)
 bool netd_control_request(uint8_t rhport, tusb_control_request_t const * request)
 {
-  // Handle class request only
-  TU_VERIFY(request->bmRequestType_bit.type == TUSB_REQ_TYPE_CLASS);
+  switch ( request->bmRequestType_bit.type )
+  {
+    case TUSB_REQ_TYPE_STANDARD:
+      switch ( request->bRequest )
+      {
+        case TUSB_REQ_GET_INTERFACE:
+          tud_control_xfer(rhport, request, &_netd_itf.itf_data_alt, 1);
+        break;
+
+        case TUSB_REQ_SET_INTERFACE:
+        {
+          uint8_t const req_itfnum = (uint8_t) request->wIndex;
+
+          // Request to enable/disable network activities on ACM-ECM only
+          TU_ASSERT(_netd_itf.ecm_mode);
+
+          // Only valid for Data Interface
+          TU_ASSERT(_netd_itf.itf_num+1 == req_itfnum);
+
+          _netd_itf.itf_data_alt = (uint8_t) request->wValue;
+
+          if ( _netd_itf.itf_data_alt )
+          {
+            // TODO since we don't actually close endpoint
+            // hack here to not re-open it
```

Comment:
Should we delay this PR until the close_endpoint PR is merged? That way we're not introducing more hacky code?

Reply:
No I think we should merge it, the dcd edpt close for all ports still has a long way to go. We also need an example to justify the change for edpt close as well. The currently is better than current master. There is many hacks already in place, I won't worry too much about those.